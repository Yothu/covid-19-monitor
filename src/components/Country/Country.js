import styled from 'styled-components';
import { useSelector } from 'react-redux';
import country from '../../assets/images/chile-silhouette.svg';
import '../../assets/Fonts.css';

const MainStats = styled.section`
  display: grid;
  grid-template-rows: 100%;
  grid-template-columns: 50% 50%;
  background-color: #5687e3;
`;

const CountryImageContainer = styled.div`

`;

const CountryDataContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

const CountryName = styled.h2`
  color: white;
  margin: 0.2rem;
  font-family: 'Gill Sans';
`;

const MainData = styled.p`
  color: white;
  display: flex;
  margin: 0 0.2rem;
  font-size: 0.9rem;
  align-items: flex-end;
  font-family: 'Lato', sans-serif;
`;

const countryStyle = {
  width: '100%',
  filter: 'invert(100)',
};

const Small = styled.span`
  margin: 0 0.2rem;
  font-size: 0.7rem;
`;

const Country = () => {
  const { totals } = useSelector((state) => state.regions);
  const CountryImage = () => <img src={country} style={countryStyle} alt="chile-silhouette" />;

  return (
    <MainStats>
      <CountryImageContainer>
        <CountryImage />
      </CountryImageContainer>
      <CountryDataContainer>
        <CountryName>{totals.name}</CountryName>
        <MainData>
          {totals.confirmed}
          <Small>confirmed</Small>
        </MainData>
        <MainData>
          {totals.deaths}
          <Small>deaths</Small>
        </MainData>
        <MainData>
          {totals.todayDeaths}
          <Small>new confirmed</Small>
        </MainData>
        <MainData>
          {totals.todayConfirmed}
          <Small>new deaths</Small>
        </MainData>
      </CountryDataContainer>
    </MainStats>

  );
};

export default Country;
