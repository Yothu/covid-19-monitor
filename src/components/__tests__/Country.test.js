import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import Country from '../Country/Country';
import store from '../../redux/configureStore';

describe('Country section tests', () => {
  it('it renders correctly', () => {
    const country = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Country />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(country).toMatchSnapshot();
  });
});
