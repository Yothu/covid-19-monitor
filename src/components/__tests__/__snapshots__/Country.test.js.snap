// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country section tests it renders correctly 1`] = `
<section
  className="sc-bdvvtL jsqFpi"
>
  <div
    className="sc-gsDKAQ YHamg"
  >
    <img
      alt="chile-silhouette"
      src="chile-silhouette.svg"
      style={
        Object {
          "filter": "invert(100)",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="sc-dkPtRN fASbqE"
  >
    <h2
      className="sc-hKwDye felzUJ"
    />
    <p
      className="sc-eCImPb ehWAiM"
    >
      <span
        className="sc-jRQBWg iHFUri"
      >
        new confirmed
      </span>
    </p>
    <p
      className="sc-eCImPb ehWAiM"
    >
      <span
        className="sc-jRQBWg iHFUri"
      >
        new deaths
      </span>
    </p>
    <p
      className="sc-eCImPb ehWAiM"
    >
      <span
        className="sc-jRQBWg iHFUri"
      >
        deaths
      </span>
    </p>
    <p
      className="sc-eCImPb ehWAiM"
    >
      <span
        className="sc-jRQBWg iHFUri"
      >
        confirmed
      </span>
    </p>
  </div>
</section>
`;
